File: python/2.py
# this will find the sum of all even numbers in the fibonacci sequesnce
#whose values do not exceed 4,000,000

def main():
    old = 1
    new = 1
    old_new = 0
    total = 0
    while new <= 4000000:
        old_new = new
        new = old + new
        if new % 2 == 0:
            total = total + new
        old = old_new
    print total
            

main()

File: javascript/2.js
// this will find the sum of all even numbers in the fibonacci sequence whose 
// values do not exceed 4,000,000
//
//answer: 4613732
//time: 0.090s

function main() {
  var x = 1
  var y = 1
  var z = 0
  var total = 0
  while (y <= 4000000) {
    z = y
    y += x
    if (y % 2 == 0) {
      total += y;
    }
    x = z
  }
  return total
}
console.log(main())
